fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailActivity.java

```diff
@@ -107,85 +114,102 @@ public void handleNewEvent(EventType eventType) {
-          getSupportFragmentManager(), meetingCreationFragment, R.id.fragment_container_lao_detail);
+              getSupportFragmentManager(), meetingCreationFragment, R.id.fragment_container_lao_detail);
     }
   }
-
   private void setupCreateRollCallFragment() {
     RollCallEventCreationFragment rollCallCreationFragment =
-        (RollCallEventCreationFragment)
-            getSupportFragmentManager().findFragmentById(R.id.fragment_create_roll_call_event);
+            (RollCallEventCreationFragment)
+                    getSupportFragmentManager().findFragmentById(R.id.fragment_create_roll_call_event);
     if (rollCallCreationFragment == null) {
       rollCallCreationFragment = RollCallEventCreationFragment.newInstance();
       ActivityUtils.replaceFragmentInActivity(
-          getSupportFragmentManager(),
-          rollCallCreationFragment,
-          R.id.fragment_container_lao_detail);
+              getSupportFragmentManager(),
+              rollCallCreationFragment,
+              R.id.fragment_container_lao_detail);
     }
   }
-
   private void setupCreatePollFragment() {
     PollEventCreationFragment pollCreationFragment =
-        (PollEventCreationFragment)
-            getSupportFragmentManager().findFragmentById(R.id.fragment_organizer_poll);
+            (PollEventCreationFragment)
+                    getSupportFragmentManager().findFragmentById(R.id.fragment_organizer_poll);
     if (pollCreationFragment == null) {
       pollCreationFragment = PollEventCreationFragment.newInstance();
       ActivityUtils.replaceFragmentInActivity(
-          getSupportFragmentManager(), pollCreationFragment, R.id.fragment_container_lao_detail);
+              getSupportFragmentManager(), pollCreationFragment, R.id.fragment_container_lao_detail);
+    }
+  }
+  private void setupScanFragmentRollCall() {
+    /*QRCodeScanningFragment scanningFragment =
+            (QRCodeScanningFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_qrcode);*/
+    QRCodeScanningFragment scanningFragment =
+            (QRCodeScanningFragment) getSupportFragmentManager().findFragmentById(R.id.add_attendee_layout);
+
+    if (scanningFragment == null) {
+      Context context = getApplicationContext();
+      BarcodeDetector qrCodeDetector = Injection.provideQRCodeDetector(context);
+      int width = getResources().getInteger(R.integer.camera_preview_width);
+      int height = getResources().getInteger(R.integer.camera_preview_height);
+      scanningFragment =
+              QRCodeScanningFragment.newInstance(
+                      Injection.provideCameraSource(context, qrCodeDetector, width, height),
+                      qrCodeDetector);
+      ActivityUtils.replaceFragmentInActivity(
+              getSupportFragmentManager(), scanningFragment, R.id.fragment_container_lao_detail);
+    }
+  }
+  private void setupCameraPermissionFragmentRollCall() {
+    CameraPermissionFragment cameraPermissionFragment =
```

Comment:
Why is it necessary to have a fragment for the cam permission ? Can't we simply have a pop up asking for permissions in the QRcodeScanningFragment ?

Reply:
This is what was done in the HomeActivity and to stay consistent I did something similar.

Reply:
This is what was done in the HomeActivity and to stay consistent I did something similar.

Reply:
There are two fragments, one for the camera permissions and the other for the QR code scanning, this way we have two corresponding View Models and we keep the logic separate